environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java

```diff
@@ -805,6 +813,45 @@ void testUpdateAzureResourceEncryptionParametersNoErrorWhenSameEncryptionKeyAlre
         verify(environmentDtoConverter, times(1)).environmentToDto(env);
     }
 
+    @Test
+    void editByNameSubnetIdChangehange() {
+        BaseNetwork awsNetwork = new AwsNetwork();
+
+        EnvironmentEditDto environmentDto = EnvironmentEditDto.builder()
+                .withAccountId(ACCOUNT_ID)
+                .withNetwork(
+                        NetworkDto.builder()
+                                .withNetworkId("abs-123")
+                                .withSubnetMetas(Map.of("subnet-1", new CloudSubnet()))
+                                .build())
+                .build();
+        Environment value = new Environment();
+        when(environmentService.findByNameAndAccountIdAndArchivedIsFalse(eq(ENVIRONMENT_NAME), eq(ACCOUNT_ID)))
+                .thenReturn(Optional.of(value));
+        when(networkService.validate(any(), any(), any()))
+                .thenReturn(awsNetwork);
+        when(networkService.refreshMetadataFromCloudProvider(any(), any(), any()))
+                .thenReturn(awsNetwork);
+        when(dnsV1Endpoint.addDnsZoneForSubnetIds(any())).thenReturn(new AddDnsZoneForSubnetsResponse());
+        when(environmentService.save(any()))
+                .thenReturn(value);
+        when(environmentDtoConverter.environmentToDto(value))
+                .thenReturn(new EnvironmentDto());
+        when(environmentDtoConverter.networkToNetworkDto(any()))
+                .thenReturn(NetworkDto.builder().withNetworkId("abs-123").build());
+
+        environmentModificationServiceUnderTest.editByName(ENVIRONMENT_NAME, environmentDto);
+
+        verify(environmentService, times(1)).findByNameAndAccountIdAndArchivedIsFalse(any(), anyString());
+        verify(networkService, times(1)).validate(any(), any(), any());
+        verify(networkService, times(1)).refreshMetadataFromCloudProvider(any(), any(), any());
+        verify(dnsV1Endpoint, times(1)).addDnsZoneForSubnetIds(any());
```

Comment:
Could you please capture the `AddDnsZoneForSubnetIdsRequest` arg and then verify its contents?

Reply:
The chain what matter here I think

Reply:
I feel it is also important that `DnsV1Endpoint` is called with the right arguments. :smile: